aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -242,11 +244,26 @@ def set_state(self, state: CalcJobState) -> None:
 
         :raise: ValueError if state is invalid
         """
+        from aiida.manage.configuration import get_config_option
+
         if not isinstance(state, CalcJobState):
             raise ValueError(f'{state} is not a valid CalcJobState')
 
         self.set_attribute(self.CALC_JOB_STATE_KEY, state.value)
 
+        if get_config_option('calcjob_state_tracking'):
+            # it is used like this in a method below, any difference with what I am doing?
+            #timestamp = timezone.datetime_to_isoformat(timezone.now())
+            timestamp = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc).isoformat()
+
+            state_tracker = self.get_attribute(self.CALC_JOB_STATE_TRACKER, None)
+            if state_tracker is None:
+                state_tracker = dict()
```

Comment:
Why don't you just do this?
```suggestion
            state_tracker = self.get_attribute(self.CALC_JOB_STATE_TRACKER, {})
```

Reply:
Hahah I wasn't super sure how this attribute system worked and I was a bit worried about using mutables for defaults (I know this is typically just for function arguments, but because the aiida entities sometimes are weird and we have had problems in the builders with this before, I was being extra careful).